### PR TITLE
Feature/serene 877 - Basic Search UI

### PR DESCRIPTION
**Pull request by dblockow-d2dcrc**
Some cleanup required but this gets us basic search functionality from a web UI.

Run the dev server with `yarn install` followed by `yarn start` from the src/main/client directory to test in dev. The maven build does this automatically and hosts it inside the spring-boot app.

search

### CODE REVIEW

File: search/pom.xml [hunk 1]
```diff
@@ -275,6 +275,51 @@
                 </configuration>
             </plugin>
 
+            <plugin>
+                <groupId>com.github.eirslett</groupId>
+                <artifactId>frontend-maven-plugin</artifactId>
+                <configuration>
+                    <workingDirectory>${basedir}/src/main/client</workingDirectory>
```
Comment (tmoschou) [inline search/pom.xml]: `${basedir}` is deprecated. should use `${project.basedir}` https://maven.apache.org/guides/introduction/introduction-to-the-pom.html#Available_Variables

File: search/src/main/client/.gitignore [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+package-lock.json*
+yarn.lock*
```
Comment (tmoschou) [inline search/src/main/client/.gitignore]: This should be checked in - see https://yarnpkg.com/lang/en/docs/yarn-lock/#toc-check-into-source-control

File: search/src/main/client/package.json [hunk 3]
```diff
@@ -0,0 +1,147 @@
+{
+  "name": "stellar-config-client",
```
Comment (tmoschou) [inline search/src/main/client/package.json]: whoops

File: search/src/main/client/src/components/edge.jsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+} from 'bloomer';
+import './searchresult.css';
+
+const Edge = (props) => {
+
+    // TODO Helper for constructing properties lists
+    let entry;
+    let edgeProperties = [];
+    for (entry in props.edge.properties) {
+        if (props.edge.properties.hasOwnProperty(entry)) {
```
Comment (tmoschou) [inline search/src/main/client/src/components/edge.jsx]: Iterating over `Object.keys(`_`obj`_`)` or `Object.entries(`_`obj`_`)` (ECMA2017) avoids needing to do the hacky `hasOwnProperty` check. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries

File: search/pom.xml (hunk 1 above)
Comment (dblockow-d2dcrc) [reply search/pom.xml]: Fixed

**Comment by dblockow-d2dcrc** [comment]
@bowernj Prod mode should work now. It's because the nav/routing causes a page refresh when changing routes and the server didn't have view controllers for `/search` and `/indexes`. They all forward to index.html now.

I'll fix routing soon but this will do in the short term.

**Pull request merged**